packages/react-devtools-shared/src/devtools/views/hooks.js

```diff
@@ -8,11 +8,75 @@
  */
 
 import throttle from 'lodash.throttle';
-import {useCallback, useEffect, useLayoutEffect, useState} from 'react';
+import {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {unstable_batchedUpdates as batchedUpdates} from 'react-dom';
 import {
   localStorageGetItem,
   localStorageSetItem,
 } from 'react-devtools-shared/src/storage';
+import {sanitizeForParse, smartParse, smartStringify} from '../utils';
+
+type EditableValue = {|
+  editableValue: any,
+  hasPendingChanges: boolean,
+  isValid: boolean,
+  parsedValue: any,
+  reset: () => void,
+  update: (newValue: any) => void,
+|};
+
+// Convenience hook for working with an editable value that is validated via JSON.parse.
+export function useEditableValue(
```

Comment:
We could potentially refactor the style editor to use this hook as well, if we replaced the inline calls to `smartStringify` and `smartParse` with injected callbacks.